### PR TITLE
Drivetrain into develop

### PR DESCRIPTION
**Pull request by JettK**
Need drivetrain gyro methods to continue field relative turret control. Drivetrain currently has basic functions done, along with odometry and kinematics.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/drivetrain/TransmissionSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  private void setTrue(){
+    m_shiftPiston.set(true);
+  }
+  
+  private void setFalse(){
+    m_shiftPiston.set(false);
+  }
+
+  @Override
```
Comment (bartkerfeld) [inline src/main/java/frc/robot/subsystems/drivetrain/TransmissionSubsystem.java:59]: Move public methods above private ones. Also consider printing the state on Shuffleboard.

File: src/main/java/frc/robot/subsystems/drivetrain/TransmissionSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    m_shiftPiston = new Solenoid(RobotMap.kDrivetrainShiftSolenoid);
+
+    m_gearState = GearState.HIGH;
+
+  }
+
+  public void setGearState(GearState state){
+    m_gearState = state;
+
+    switch(state){
```
Comment (bartkerfeld) [inline src/main/java/frc/robot/subsystems/drivetrain/TransmissionSubsystem.java]: This format looks a little weird. Please run Format Document on this file.

File: src/main/java/frc/robot/subsystems/drivetrain/DrivetrainSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    m_rightSlave = new WPI_TalonFX(RobotMap.kDrivetrainRightFrontTalonFX);
+
+    m_pigeon = new PigeonIMU(RobotMap.kPigeonIMU);
+
+    //Setting followers, followers don't automatically follow master's inverts so you must set the invert type to FollowMaster
+    m_leftSlave.follow(m_leftMaster, FollowerType.PercentOutput);
+    m_leftSlave.setInverted(InvertType.FollowMaster);
+    m_rightSlave.follow(m_rightMaster, FollowerType.PercentOutput);
+    m_rightSlave.setInverted(InvertType.FollowMaster);
+
```
Comment (bartkerfeld) [inline src/main/java/frc/robot/subsystems/drivetrain/DrivetrainSubsystem.java:61]: You will also need to invert the right side speed controllers. I don't see that yet.

File: src/main/java/frc/robot/subsystems/drivetrain/DrivetrainSubsystem.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+       */
+      fx.configSupplyCurrentLimit(new SupplyCurrentLimitConfiguration(true, 45, 80, 30));
+
+      //Either using the integrated Falcon sensor or an external one, will change if needed
+      fx.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor);
+    }
+
+    resetEncoders();
+
+    differentialdrive = new DifferentialDrive(m_leftMaster, m_rightMaster);
```
Comment (bartkerfeld) [inline src/main/java/frc/robot/subsystems/drivetrain/DrivetrainSubsystem.java:97]: You will also need to call differentialdrive.setRightSideInverted(false)

File: src/main/java/frc/robot/subsystems/drivetrain/DrivetrainSubsystem.java (hunk 3 above)
Comment (JettK) [reply src/main/java/frc/robot/subsystems/drivetrain/DrivetrainSubsystem.java:61]: Fixed

File: src/main/java/frc/robot/subsystems/drivetrain/DrivetrainSubsystem.java (hunk 4 above)
Comment (JettK) [reply src/main/java/frc/robot/subsystems/drivetrain/DrivetrainSubsystem.java:97]: Fixed

**Pull request merged**